src/main/java/org/jabref/gui/consistency/ConsistencyCheckAction.java

```diff
@@ -0,0 +1,53 @@
+package org.jabref.gui.consistency;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.preferences.GuiPreferences;
+import org.jabref.logic.quality.consistency.BibliographyConsistencyCheck;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+
+public class ConsistencyCheckAction extends SimpleCommand {
+
+    private final DialogService dialogService;
+    private final StateManager stateManager;
+    private final GuiPreferences preferences;
+
+    private final List<String> entryTypes = new ArrayList<>();
+    private final List<String> citationKeys = new ArrayList<>();
+    private final List<String> columns = new ArrayList<>();
+
+    public ConsistencyCheckAction(DialogService dialogService, StateManager stateManager, GuiPreferences preferences) {
+        this.dialogService = dialogService;
+        this.stateManager = stateManager;
+        this.preferences = preferences;
+    }
+
+    @Override
+    public void execute() {
+        BibDatabaseContext databaseContext = stateManager.getActiveDatabase().orElseThrow(() -> new NullPointerException("Database null"));
+        List<BibEntry> entries = databaseContext.getDatabase().getEntries();
+
+        BibliographyConsistencyCheck consistencyCheck = new BibliographyConsistencyCheck();
+        BibliographyConsistencyCheck.Result result = consistencyCheck.check(entries);
```

Comment:
We should probably run this in a background thread to avoid blocking the UI, I'm afraid it would be noticeable in large databases. Please learn about `BackgroundTask` and how it is used in other parts of the codebase.